### PR TITLE
Update nc dependency to 0.9.2

### PR DESCRIPTION
**Pull request by posutsai**
Hi, new to youki here. Hope to start my contribution from a simple issue #2874. 

1. Instead of updating to 0.9.1, I update to 0.9.2
2. Since the `size` parameter is needless, I remove it.

Please let me know if I do anything wrong. Thank you and hope to do more contribution.

### CODE REVIEW

**Comment by YJDoc2** [comment]
Hey the lint CI is failing - https://github.com/containers/youki/actions/runs/10454851896/job/28979505684?pr=2884 . Also, you will need to resolve conflicts due to another  dependabot pr getting merged.

File: tests/contest/runtimetest/Cargo.toml [hunk 1]
```diff
@@ -7,5 +7,5 @@ edition = "2021"
 oci-spec = { version = "0.6.8", features = ["runtime"] }
 nix = "0.28.0"
 anyhow = "1.0"
-libc = "0.2.158" # TODO (YJDoc2) upgrade to latest
-nc = "0.8.23"
+libc = "0.2.157" # TODO (YJDoc2) upgrade to latest
```
Comment (YJDoc2) [inline tests/contest/runtimetest/Cargo.toml:10]: You have accidentally set the libc version to older one, please only change the nc version.

**Review by YJDoc2 — CHANGES_REQUESTED** [review]
one minor change as a result of conflict resolve, otherwise ok.

File: tests/contest/runtimetest/Cargo.toml (hunk 1 above)
Comment (posutsai) [reply tests/contest/runtimetest/Cargo.toml]: Sorry for that. I already fix it. Thank you.

**Review by YJDoc2 — APPROVED** [review]
lgtm! Thanks for contributing!

**Pull request merged**